ecosystem/indexer-grpc/indexer-grpc-table-info/src/table_info_service.rs

```diff
@@ -330,3 +468,96 @@ impl TableInfoService {
         Ok(ledger_version)
     }
 }
+
+async fn backup_the_snapshot_and_cleanup(
+    context: Arc<ApiContext>,
+    backup_restore_operator: Arc<GcsBackupRestoreOperator>,
+    epoch: u64,
+) {
+    let snapshot_folder_name = snapshot_folder_name(context.chain_id().id() as u64, epoch);
+
+    let ledger_chain_id = context.chain_id().id();
+    // Validate the runtime.
+    let backup_metadata = backup_restore_operator.get_metadata().await;
+    if let Some(metadata) = backup_metadata {
+        if metadata.chain_id != (ledger_chain_id as u64) {
+            panic!(
+                "Table Info backup chain id does not match with current network. Expected: {}, found in backup: {}",
+                context.chain_id().id(),
+                metadata.chain_id
+            );
+        }
+    }
+
+    let start_time = std::time::Instant::now();
+    // temporary path to store the snapshot
+    let snapshot_dir = context
+        .node_config
+        .get_data_dir()
+        .join(snapshot_folder_name);
+
+    // If the backup is for old epoch, clean up and return.
+    if let Some(metadata) = backup_metadata {
+        if metadata.epoch >= epoch {
+            // Remove the snapshot directory.
+            std::fs::remove_dir_all(snapshot_dir).unwrap();
+            return;
+        }
+    }
+
+    backup_restore_operator
```

Comment:
I'm not sure if you need to do something here to handle the case when multiple jobs uploading the backup at the same time. Can you double check?

Reply:
Do you mean multiple backup services are running? No; my thought is to check it on the deployment side. If we do want to check within the program(runtime?), we can either use some distributed locking or some coarse method like, checking metadata versioning. wdyt?

Reply:
I feel we want to have multiple jobs to be able to backup things to ensure a fair reliability. I don't have preference on how they corporate. Either doing it on client side or on server side are ok.
We can leave it as a TODO for future PR.

Reply:
(I'd just add a random suffix to the file name so we never write to the same remote file and we can observe the conflicting behavior better)

Reply:
Thanks! I added a to-do as follow-up to check if concurrent jobs are running. One of the concurrent scenarios for now is multiple full nodes running with backup enabled.